src/shared/components/unsupportedBrowserModal/UnsupportedBrowserModal.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { Modal } from 'react-bootstrap';
+import browser from 'detect-browser';
+
+
+interface IBrowserState {
+    show: boolean;
+    name: string;
+    version: string;
+}
+
+export default class UnsupportedBrowserModal extends React.Component<{}, IBrowserState> {
+
+  constructor() {
+      super();
+
+      if (!browser)  {
+          this.state = {
+              name: "Unsupported",
+              show: true,
+              version: ""
+          };
+      } else {
+          this.state = {
+              name: browser.name,
+              show: false,
+               version: browser.version
+          };
+      }
+  }
+
+  componentDidMount() {
+      this.handleUnsupportedBrowsers(this.state.name, this.state.version);
+  }
+
+  handleUnsupportedBrowsers(name:string, version: string) {
+      name = name.toLowerCase();
+      const localStorage = window.localStorage.browserError || false;
+      if (localStorage === true) {
+          this.setState({show: false});
+      } else if (String(name) === 'unsupported') {
```

Comment:
Ino told me to add it in the case that `detect-browser` could not detect the browser. Up in the constructor, if `browser` is undefined, I set the browser state to be `unsupported`.